app/src/main/java/cl/mario/covid/util/CalendarManager.kt

```diff
@@ -0,0 +1,37 @@
+package cl.mario.covid.util
+
+import android.app.DatePickerDialog
+import cl.mario.covid.ui.main.MainActivity
+import java.util.*
+
+class CalendarManager(private val activity: MainActivity) {
+
+    private var dateSelectedListener: ((date: Date) -> Unit)? = null
```

Comment:
It not recommended to pass an activity reference (It's better to pass a little of information), maybe you could use context  :)